src/vm/clarity.rs

```diff
@@ -841,9 +841,14 @@ impl<'a, 'b> ClarityTransactionConnection<'a, 'b> {
         from: &PrincipalData,
         to: &PrincipalData,
         amount: u128,
+        memo: &ASCIIData,
```

Comment:
I think the memo in regular stx-transfer transactions is an arbitrary 34-byte array (e.g. not constrained to ASCII characters). This should be a buffer object rather than an ascii string.